src/agent/src/workload/runner.rs

```diff
@@ -19,7 +19,7 @@ pub struct Runner {
 impl Runner {
     pub fn new(config: Config) -> Self {
         let agent: Box<dyn Agent + Sync + Send> = match config.language {
-            Language::Rust => Box::new(rust::RustAgent::from(config.clone())),
+            Language::Rust => Box::new(rust::RustAgent::try_from(config.clone()).expect("Failed to convert Config to RustAgent")),
```

Comment:
`expect()` panics so it's like an `unwrap()`

Reply:
That's true. I wanted to do something like this :
```rust
let agent = match rust::RustAgent::try_from(config.clone()) {
    Ok(agent) => Box::new(agent),
    Err(err) => {
       // I don't know how to handle the error here as propagating is not an option
    }
};
```

Reply:
so we need to return a result for `Runner::new()`

and you don't need the `match` if the `RustAgent` `try_from` already returns an `AgentError` (to change in the Agent also)